File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>st;
        vector<int>vc;
        int n=nums.size();
        for(int i=0;i<n;i++)
        {
            int remai = target-nums[i];
            if(st.find(remai)!=st.end())
            {
                vc.push_back(st[remai]);
                vc.push_back(i);
            }
            st[nums[i]]=i;
        }
        return vc;
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] arr = new int[2];
        HashMap<Integer,Integer>st = new HashMap<>();
        for(int i=0;i<nums.length;i++)
        {
           int rem = target-nums[i];
           if(st.containsKey(rem))
           {
            arr[0]=st.get(rem);
            arr[1]=i;
            return arr;
           }
           st.put(nums[i],i);
        }
        return arr;
    }
}